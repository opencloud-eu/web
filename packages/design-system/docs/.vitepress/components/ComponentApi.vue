<template>
  <h2 id="properties" tabindex="-1">
    Properties
    <a class="header-anchor" href="#properties" aria-label='Permalink to "Properties"'>​</a>
  </h2>
  <div v-if="isLoading" class="oc-flex oc-flex-center">
    <oc-spinner aria-label="Loading properties" />
  </div>
  <div v-else>
    <oc-table v-if="props.length" :fields="fields" :data="props" class="props-table">
      <template #type="{ item }">
        <code> {{ item.type }}</code>
      </template>
      <template #description="{ item }">
        <span v-html="item.description.replace(/`([^`]+)`/g, '<code>$1</code>')" />
        <span v-if="item.default">
          Defaults to <code>{{ item.default }}</code
          >.</span
        >
      </template>
    </oc-table>
    <div v-else>This component doesn't define properties.</div>
  </div>
  <h2 id="emits" tabindex="-1">
    Emits
    <a class="header-anchor" href="#emits" aria-label='Permalink to "Emits"'>​</a>
  </h2>
  <div v-if="isLoading" class="oc-flex oc-flex-center">
    <oc-spinner aria-label="Loading emits" />
  </div>
  <div v-else>
    <oc-table
      v-if="emits.length"
      :fields="fields.filter((f) => f.name !== 'required')"
      :data="emits"
      class="emits-table"
    >
      <template #type="{ item }">
        <code> {{ item.type }}</code>
      </template>
      <template #description="{ item }">
        <span v-html="item.description.replace(/`([^`]+)`/g, '<code>$1</code>')" />
      </template>
    </oc-table>
    <div v-else>This component doesn't define emits.</div>
  </div>
  <h2 id="slots" tabindex="-1">
    Slots
    <a class="header-anchor" href="#slots" aria-label='Permalink to "Slots"'>​</a>
  </h2>
  <div v-if="isLoading" class="oc-flex oc-flex-center">
    <oc-spinner aria-label="Loading slots" />
  </div>
  <div v-else>
    <oc-table
      v-if="slots.length"
      :fields="fields.filter((f) => f.name !== 'required')"
      :data="slots"
      class="slots-table"
    >
      <template #type="{ item }">
        <code> {{ item.type }}</code>
      </template>
      <template #description="{ item }">
        <span v-html="item.description.replace(/`([^`]+)`/g, '<code>$1</code>')" />
      </template>
    </oc-table>
    <div v-else>This component doesn't define slots.</div>
  </div>
</template>

<script setup lang="ts">
import { ref, onMounted, Ref, unref } from 'vue'
import { Emit, Prop, Slot } from '../generateJsonMetaData'
import { useComponentName } from '../composables/useComponentName'

const { component } = defineProps<{ component?: string }>()

const { componentName } = useComponentName()

const fields = [
  {
    name: 'name',
    title: 'Name'
  },
  {
    name: 'required',
    title: 'Required'
  },
  {
    name: 'type',
    title: 'Type',
    type: 'slot'
  },
  {
    name: 'description',
    title: 'Description',
    type: 'slot'
  }
]

const emits: Ref<Emit[]> = ref([])
const props: Ref<Prop[]> = ref([])
const slots: Ref<Slot[]> = ref([])
const isLoading = ref(true)

const load = async (type: string, componentName: string) => {
  try {
    const response = await fetch(`../${type}/${componentName}.json`)
    if (response.headers?.get('content-type')?.startsWith('application/json')) {
      return response.json()
    }
    return []
  } catch (e) {
    console.error(`loading ${type}:`, e)
  }
}

onMounted(async () => {
  emits.value = await load('emits', component || unref(componentName))
  props.value = await load('props', component || unref(componentName))
  slots.value = await load('slots', component || unref(componentName))
  isLoading.value = false
})
</script>

<style lang="scss">
.emits-table,
.props-table,
.slots-table {
  display: table !important;
  width: 100%;
  th,
  td {
    background-color: #fff;
    border: 0;
    border-bottom: 1px solid var(--vp-c-divider);
  }
  tr {
    border: 0;
  }
}
</style>
