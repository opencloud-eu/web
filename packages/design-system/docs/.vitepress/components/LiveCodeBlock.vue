<template>
  <div class="live-code-block oc-mt-l oc-mb-xl">
    <div class="live-code-block-header oc-mb-m">
      <oc-button
        @click="() => (previewActive = true)"
        appearance="raw"
        :class="{ active: previewActive }"
        class="oc-px-m oc-pb-s"
        no-hover
        >Preview</oc-button
      >
      <oc-button
        @click="() => (previewActive = false)"
        appearance="raw"
        :class="{ active: !previewActive }"
        class="oc-px-m oc-pb-s"
        no-hover
        >Code</oc-button
      >
    </div>
    <div :class="{ 'oc-hidden': !previewActive }">
      <component :is="preview" />
    </div>
    <div ref="slot" :class="{ 'oc-hidden': previewActive }">
      <slot />
    </div>
  </div>
</template>

<script setup lang="ts">
import { CompilerError } from '@vue/compiler-dom'
import { computed, unref, useTemplateRef, compile, ref, onMounted } from 'vue'

const { path } = defineProps<{
  path?: string // path to an external code example component. needs to be relative to the docs root folder.
}>()

const slot = useTemplateRef<HTMLElement>('slot')
const preview = ref()
const previewActive = ref(true)

const components = import.meta.glob('./../../components/**/*.vue')

const computedPath = computed(() => {
  if (!path) {
    return ''
  }
  // file extension needs to be removed and re-added manually to avoid compiler warnings
  const previewPath = path.startsWith('/') ? path.slice(1) : path
  const fileExtension = previewPath.split('.').pop()
  const previewPathWithoutExtension = previewPath.slice(0, -(fileExtension?.length || 1) - 1)
  return `../../${previewPathWithoutExtension}.${fileExtension}`
})

const lang = computed(() => {
  // extracts language from code block
  const htmlStr = unref(slot)?.innerHTML
  if (!htmlStr) {
    return ''
  }
  const regex = /<span class="lang">([^<]+)<\/span>/
  const match = htmlStr.match(regex)
  if (match) {
    return match[1]
  }
  return ''
})

onMounted(async () => {
  if (path && components[unref(computedPath)]) {
    const previewComponent = await components[unref(computedPath)]()
    preview.value = previewComponent.default
    return
  }

  const textContent = unref(slot)?.textContent
  if (!textContent) {
    return ''
  }
  const templateStr = textContent.substring(unref(lang).length)
  preview.value = compile(templateStr, {
    whitespace: 'preserve',
    onError: (error: CompilerError) => {
      console.error(error)
    }
  })
})
</script>

<style lang="scss" scoped>
.live-code-block-header {
  border-bottom: 1px solid var(--vp-c-divider);
  button {
    border-radius: 0;
  }
  button.active {
    border-bottom: 2px solid var(--vp-c-brand-1);
  }
}
</style>
