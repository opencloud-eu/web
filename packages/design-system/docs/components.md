### Components are the smallest basic structures of an user interface. They cannot be broken down any further. Buttons, links, and inputs are good examples. Components utilize decisions made on the design token level.

You’re looking at OpenCloud Design System’s demo <code>UI Components</code>. Everything you see here is editable in Markdown format. To change or remove this content, see [/docs/components.md](https://github.com/opencloud-eu/web/blob/main/packages/design-system/docs/components.md). The documentation you see below is auto generated by Vue Styleguidist. To edit how it’s formatted, see [/docs/docs.styles.scss](https://github.com/opencloud-eu/web/blob/main/packages/design-system/docs/docs.styles.scss).

## Overview

```
<docs-components-list />
```
