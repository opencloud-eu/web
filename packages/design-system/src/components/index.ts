export { default as OcApplicationIcon } from './OcApplicationIcon/OcApplicationIcon.vue'
export { default as OcAvatar } from './OcAvatar/OcAvatar.vue'
export { default as OcAvatarCount } from './OcAvatarCount/OcAvatarCount.vue'
export { default as OcAvatarFederated } from './OcAvatarFederated/OcAvatarFederated.vue'
export { default as OcAvatarGroup } from './OcAvatarGroup/OcAvatarGroup.vue'
export { default as OcAvatarGuest } from './OcAvatarGuest/OcAvatarGuest.vue'
export { default as OcAvatarItem } from './OcAvatarItem/OcAvatarItem.vue'
export { default as OcAvatarLink } from './OcAvatarLink/OcAvatarLink.vue'
export { default as OcAvatars } from './OcAvatars/OcAvatars.vue'
export { default as OcBreadcrumb } from './OcBreadcrumb/OcBreadcrumb.vue'
export { default as OcButton } from './OcButton/OcButton.vue'
export { default as OcCheckbox } from './OcCheckbox/OcCheckbox.vue'
export { default as OcColorInput } from './OcColorInput/OcColorInput.vue'
export { default as OcContextualHelper } from './OcContextualHelper/OcContextualHelper.vue'
export { default as OcDatepicker } from './OcDatepicker/OcDatepicker.vue'
export { default as OcDefinitionList } from './OcDefinitionList/OcDefinitionList.vue'
export { default as OcDrop } from './OcDrop/OcDrop.vue'
export { default as OcDropzone } from './OcDropzone/OcDropzone.vue'
export { default as OcFileInput } from './OcFileInput/OcFileInput.vue'
export { default as OcFilterChip } from './OcFilterChip/OcFilterChip.vue'
export { default as OcGrid } from './OcGrid/OcGrid.vue'
export { default as OcHiddenAnnouncer } from './OcHiddenAnnouncer/OcHiddenAnnouncer.vue'
export { default as OcIcon } from './OcIcon/OcIcon.vue'
export { default as OcImage } from './OcImage/OcImage.vue'
export { default as OcInfoDrop } from './OcInfoDrop/OcInfoDrop.vue'
export { default as OcList } from './OcList/OcList.vue'
export { default as OcLoader } from './OcLoader/OcLoader.vue'
export { default as OcModal } from './OcModal/OcModal.vue'
export { default as OcNotificationMessage } from './OcNotificationMessage/OcNotificationMessage.vue'
export { default as OcNotifications } from './OcNotifications/OcNotifications.vue'
export { default as OcPageSize } from './OcPageSize/OcPageSize.vue'
export { default as OcPagination } from './OcPagination/OcPagination.vue'
export { default as OcProgress } from './OcProgress/OcProgress.vue'
export { default as OcProgressPie } from './OcProgressPie/OcProgressPie.vue'
export { default as OcRadio } from './OcRadio/OcRadio.vue'
export { default as OcRecipient } from './OcRecipient/OcRecipient.vue'
export { default as OcSearchBar } from './OcSearchBar/OcSearchBar.vue'
export { default as OcSelect } from './OcSelect/OcSelect.vue'
export { default as OcSpinner } from './OcSpinner/OcSpinner.vue'
export { default as OcStatusIndicators } from './OcStatusIndicators/OcStatusIndicators.vue'
export { default as OcSwitch } from './OcSwitch/OcSwitch.vue'
export { default as OcTable } from './OcTable/OcTable.vue'
export { default as OcTableBody } from './OcTableBody/OcTableBody.vue'
export { default as OcTableFoot } from './OcTableFoot/OcTableFoot.vue'
export { default as OcTableHead } from './OcTableHead/OcTableHead.vue'
export { default as OcTableSimple } from './OcTableSimple/OcTableSimple.vue'
export { default as OcTableTd } from './OcTableTd/OcTableTd.vue'
export { default as OcTableTh } from './OcTableTh/OcTableTh.vue'
export { default as OcTableTr } from './OcTableTr/OcTableTr.vue'
export { default as OcTag } from './OcTag/OcTag.vue'
export { default as OcTextarea } from './OcTextarea/OcTextarea.vue'
export { default as OcTextInput } from './OcTextInput/OcTextInput.vue'
export { default as OcErrorLog } from './OcErrorLog/OcErrorLog.vue'
export { default as OcEmojiPicker } from './OcEmojiPicker/OcEmojiPicker.vue'
