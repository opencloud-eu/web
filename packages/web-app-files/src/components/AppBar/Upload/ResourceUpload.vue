<template>
  <div
    v-oc-tooltip="
      isRemoteUploadInProgress ? $gettext('Please wait until all imports have finished') : null
    "
  >
    <oc-button
      :class="btnClass"
      justify-content="left"
      appearance="raw"
      :disabled="isRemoteUploadInProgress"
      @click="triggerUpload"
    >
      <resource-icon :resource="resource" size="medium" />
      <span :id="uploadLabelId">{{ buttonLabel }}</span>
    </oc-button>
    <input
      :id="inputId"
      ref="input"
      v-bind="inputAttrs"
      class="upload-input-target"
      type="file"
      :aria-labelledby="uploadLabelId"
      :name="isFolder ? 'file' : 'folder'"
      tabindex="-1"
    />
  </div>
</template>

<script lang="ts">
import {
  computed,
  defineComponent,
  onMounted,
  onBeforeUnmount,
  ref,
  unref,
  useTemplateRef
} from 'vue'
import { Resource } from '@opencloud-eu/web-client'
import { useService, ResourceIcon, convertToMinimalUppyFile } from '@opencloud-eu/web-pkg'
import type { UppyService } from '@opencloud-eu/web-pkg'
import { getItemsViaDirectoryPicker } from '../../../helpers/directoryPicker'

export default defineComponent({
  components: { ResourceIcon },
  props: {
    btnLabel: {
      type: String,
      required: false,
      default: ''
    },
    btnClass: {
      type: String,
      required: false,
      default: ''
    },
    isFolder: {
      type: Boolean,
      required: false,
      default: false
    }
  },
  setup(props) {
    const uppyService = useService<UppyService>('$uppyService')
    const input = useTemplateRef<HTMLInputElement>('input')

    const isRemoteUploadInProgress = ref(uppyService.isRemoteUploadInProgress())

    let uploadStartedSub: string
    let uploadCompletedSub: string

    const resource = computed(() => {
      return { extension: '', isFolder: props.isFolder } as Resource
    })

    const onUploadStarted = () =>
      (isRemoteUploadInProgress.value = uppyService.isRemoteUploadInProgress())
    const onUploadCompleted = () => (isRemoteUploadInProgress.value = false)

    const triggerUpload = async () => {
      if (!props.isFolder || typeof (window as any).showDirectoryPicker !== 'function') {
        // use native file picker for file uploads or if browser does not support the Directory API
        unref(input).click()
        return
      }

      try {
        // use the Directory API so we can retrieve empty folders
        const items = await getItemsViaDirectoryPicker((error) => uppyService.log(error))
        const uppyFiles = convertToMinimalUppyFile('FolderUpload', items)
        uppyService.addFiles(uppyFiles)
      } catch (error) {
        if (error.name !== 'AbortError') {
          // AbortError means the user closed the picker. in any other case
          // we assume something went wrong and we fall back to the native picker.
          console.error('Error using DirectoryPicker, falling back to the native one:', error)
          unref(input).click()
        }
      }
    }

    onMounted(() => {
      uploadStartedSub = uppyService.subscribe('uploadStarted', onUploadStarted)
      uploadCompletedSub = uppyService.subscribe('uploadCompleted', onUploadCompleted)
      uppyService.registerUploadInput(unref(input))
    })

    onBeforeUnmount(() => {
      uppyService.unsubscribe('uploadStarted', uploadStartedSub)
      uppyService.unsubscribe('uploadCompleted', uploadCompletedSub)
      uppyService.removeUploadInput(unref(input))
    })
    return {
      isRemoteUploadInProgress,
      resource,
      input,
      triggerUpload
    }
  },
  computed: {
    inputId() {
      if (this.isFolder) {
        return 'files-folder-upload-input'
      }
      return 'files-file-upload-input'
    },
    uploadLabelId() {
      if (this.isFolder) {
        return 'files-folder-upload-button'
      }
      return 'files-file-upload-button'
    },
    buttonLabel() {
      if (this.btnLabel) {
        return this.btnLabel
      }
      if (this.isFolder) {
        return this.$gettext('Folder')
      }
      return this.$gettext('Files')
    },
    inputAttrs() {
      if (this.isFolder) {
        return {
          webkitdirectory: true,
          mozdirectory: true,
          allowdirs: true
        }
      }
      return { multiple: true }
    }
  }
})
</script>

<style scoped>
.upload-input-target {
  position: absolute;
  left: -99999px;
}
</style>
