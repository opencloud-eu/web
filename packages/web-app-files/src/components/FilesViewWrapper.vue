<template>
  <div class="files-view-wrapper grid grid-cols-1 flex-1 focus:outline-0 h-full overflow-y-auto">
    <div id="files-view" v-bind="$attrs" class="outline-0">
      <slot />
    </div>
  </div>

  <portal v-if="isEmbedModeEnabled" to="app.runtime.footer">
    <embed-actions />
  </portal>
</template>

<script lang="ts">
import { defineComponent } from 'vue'
import { useEmbedMode } from '@opencloud-eu/web-pkg'
import EmbedActions from './EmbedActions/EmbedActions.vue'

export default defineComponent({
  components: { EmbedActions },
  inheritAttrs: false,
  setup() {
    const { isEnabled: isEmbedModeEnabled } = useEmbedMode()

    return { isEmbedModeEnabled }
  }
})
</script>

<style lang="scss" scoped>
.files-view-wrapper {
  position: relative;
  grid-template-rows: max-content max-content 1fr;
  gap: 0 0;
  grid-template-areas:
    'header'
    'upload'
    'main';
}

#files-view {
  grid-area: main;
  z-index: 0;
}
</style>
