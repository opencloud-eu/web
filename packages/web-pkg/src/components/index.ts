export * from './AppBar'
export * from './AppTemplates'
export * from './ContextActions'
export * from './FilesList'
export * from './Filters'
export * from './Modals'
export * from './SideBar'
export * from './Search'
export * from './Spaces'
export * from './TextEditor'

export { default as AppLoadingSpinner } from './AppLoadingSpinner.vue'
export { default as AppTopBar } from './AppTopBar.vue'
export { default as BatchActions } from './BatchActions.vue'
export { default as CustomComponentTarget } from './CustomComponentTarget.vue'
export { default as ItemFilter } from './ItemFilter.vue'
export { default as ItemFilterToggle } from './ItemFilterToggle.vue'
export { default as LinkRoleDropdown } from './LinkRoleDropdown.vue'
export { default as LoadingIndicator } from './LoadingIndicator.vue'
export { default as NoContentMessage } from './NoContentMessage.vue'
export { default as Pagination } from './Pagination.vue'
export { default as QuotaSelect } from './QuotaSelect.vue'
export { default as SpaceQuota } from './SpaceQuota.vue'
export { default as SearchBarFilter } from './SearchBarFilter.vue'
export { default as ViewOptions } from './ViewOptions.vue'
export { default as PortalTarget } from './PortalTarget.vue'
export { default as CreateShortcutModal } from './CreateShortcutModal.vue'
export { default as CreateLinkModal } from './CreateLinkModal.vue'
