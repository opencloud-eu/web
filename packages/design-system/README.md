# OpenCloud Design System

[![Matrix](https://img.shields.io/matrix/opencloud%3Amatrix.org?logo=matrix)](https://app.element.io/#/room/#opencloud:matrix.org)

The **OpenCloud Design System** provides components and utilities for application and extension development within the
OpenCloud Web ecosystem. It can be developed standalone via the design system documentation. The documentation is
built with [VitePress](https://vitepress.dev/).

Head over to the [hosted docs](https://docs.opencloud.eu/design-system/) for more information!

## Running the docs locally

To run the docs, you first need to install all dependencies:

```
pnpm i
```

Start docs in dev mode:

```
pnpm docs:dev
```

## Usage as a package

### Installation

To use the design-system in your application, you first need to install the package. Depending on your package manager, run one of the following commands:

```
$ npm install @opencloud-eu/design-system

$ pnpm add @opencloud-eu/design-system

$ yarn add @opencloud-eu/design-system
```

Note that if you're using the design-system in an OpenCloud Web app, it's recommended to install it as dev dependency. This is because the Web runtime already ships the design-system and you only need it for development purposes in your IDE.

### Styles

In order to use the provided CSS classes and to ensure the components are styled correctly, you need to import the styles like so:

```
import '@opencloud-eu/design-system/dist/design-system.css'
```

Again, this is not needed if you're using the design-system in an OpenCloud Web app because the styles are already available via the Web runtime.

### Components

To use a component, you need to import it. For example, to use the `OcButton` component:

```
import { OcButton } from '@opencloud-eu/design-system/components'

<oc-button>
  Click me!
</oc-button>
```

You can also register the components globally in your standalone Vue app. This way you don't need to import them any time you want to use them.

```
import { createApp } from 'vue'
import DesignSystem from '@opencloud-eu/design-system'

const app = createApp({ ... })
app.use(DesignSystem)
```

In order for your IDE to pick up the correct component types, you need to add the following to your `types.d.ts` file:

```
/// <reference types="@opencloud-eu/design-system/types" />
```

Optionally, you can pass custom design tokens to the design-system. Check the [example theme](https://github.com/opencloud-eu/opencloud/blob/v2.2.0/services/web/assets/themes/opencloud/theme.json) for a list of available tokens.

```
const tokens = {
  spacing: {
    small: '4px',
    medium: '8px',
    large: '16px',
  }
}

app.use(DesignSystem, { tokens })
```
