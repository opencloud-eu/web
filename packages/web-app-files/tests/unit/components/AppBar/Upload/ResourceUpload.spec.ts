import { mockDeep } from 'vitest-mock-extended'
import ResourceUpload from '../../../../../src/components/AppBar/Upload/ResourceUpload.vue'
import {
  defaultComponentMocks,
  defaultPlugins,
  defaultStubs,
  mount
} from '@opencloud-eu/web-test-helpers'
import { UppyService } from '@opencloud-eu/web-pkg'
import { OcButton } from '@opencloud-eu/design-system/components'

describe('Resource Upload Component', () => {
  describe('file upload', () => {
    it('should render component', () => {
      const { wrapper } = getWrapper()
      expect(wrapper.exists()).toBeTruthy()
      expect(wrapper.html()).toMatchSnapshot()
    })
  })

  describe('folder upload', () => {
    it('should render component', () => {
      const { wrapper } = getWrapper({ isFolder: true })
      expect(wrapper.exists()).toBeTruthy()
      expect(wrapper.html()).toMatchSnapshot()
    })
  })

  describe('when upload file button is clicked', () => {
    it('should call "triggerUpload"', async () => {
      const { wrapper } = getWrapper()

      const spyTriggerUpload = vi.spyOn(wrapper.vm, 'triggerUpload')
      const uploadButton = wrapper.find('button')
      const fileUploadInput = wrapper.find('#files-file-upload-input')

      ;(fileUploadInput.element as HTMLElement).click = vi.fn()
      await wrapper.vm.$forceUpdate()

      await uploadButton.trigger('click')

      expect(spyTriggerUpload).toHaveBeenCalledTimes(1)
      expect((fileUploadInput.element as HTMLElement).click).toHaveBeenCalledTimes(1)
    })
  })

  it('should be disabled when a remote upload is running', () => {
    const uppyService = mockDeep<UppyService>()
    uppyService.isRemoteUploadInProgress.mockReturnValue(true)
    const { wrapper } = getWrapper({ isFolder: true }, uppyService)
    expect(wrapper.findComponent<typeof OcButton>('button').props('disabled')).toBeTruthy()
  })
})

function getWrapper(props = {}, uppyService = mockDeep<UppyService>()) {
  const mocks = {
    ...defaultComponentMocks(),
    $uppyService: uppyService
  }
  return {
    mocks,
    wrapper: mount(ResourceUpload, {
      props,
      global: {
        mocks,
        stubs: defaultStubs,
        provide: mocks,
        plugins: [...defaultPlugins()]
      }
    })
  }
}
