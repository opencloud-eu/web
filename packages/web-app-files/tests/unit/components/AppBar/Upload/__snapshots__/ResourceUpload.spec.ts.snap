// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Resource Upload Component > file upload > should render component 1`] = `
"<div data-v-4a3fd5f0=""><button data-v-4a3fd5f0="" type="button" class="oc-button oc-rounded oc-button-m oc-button-justify-content-left oc-button-gap-m oc-button-secondary oc-button-raw oc-button-secondary-raw">
    <!--v-if-->
    <!-- @slot Content of the button --> <span data-v-4a3fd5f0="" class="oc-icon oc-icon-m oc-resource-icon oc-resource-icon-file"><!----></span> <span data-v-4a3fd5f0="" id="files-file-upload-button">Files</span>
  </button> <input data-v-4a3fd5f0="" id="files-file-upload-input" multiple="" class="upload-input-target" type="file" aria-labelledby="files-file-upload-button" name="folder" tabindex="-1"></div>"
`;

exports[`Resource Upload Component > folder upload > should render component 1`] = `
"<div data-v-4a3fd5f0=""><button data-v-4a3fd5f0="" type="button" class="oc-button oc-rounded oc-button-m oc-button-justify-content-left oc-button-gap-m oc-button-secondary oc-button-raw oc-button-secondary-raw">
    <!--v-if-->
    <!-- @slot Content of the button --> <span data-v-4a3fd5f0="" class="oc-icon oc-icon-m oc-resource-icon oc-resource-icon-folder"><!----></span> <span data-v-4a3fd5f0="" id="files-folder-upload-button">Folder</span>
  </button> <input data-v-4a3fd5f0="" id="files-folder-upload-input" webkitdirectory="true" mozdirectory="true" allowdirs="true" class="upload-input-target" type="file" aria-labelledby="files-folder-upload-button" name="file" tabindex="-1"></div>"
`;
